app/uk/gov/hmrc/agentsubscriptionfrontend/models/RadioWithInput.scala

```diff
@@ -18,18 +18,29 @@ package uk.gov.hmrc.agentsubscriptionfrontend.models
 
 import play.api.data.Form
 import play.api.data.Forms._
-import uk.gov.hmrc.agentsubscriptionfrontend.controllers.FieldMappings.radioInputSelected
+import uk.gov.hmrc.agentsubscriptionfrontend.controllers.FieldMappings.{nino, radioInputSelected, saAgentCode, utr}
+import uk.gov.voa.play.form.ConditionalMappings.{mandatoryIfEqual, mandatoryIfTrue}
 
-case class RadioWithInput(
-  value: Option[Boolean],
-  messageOfTrueRadioChoice: Option[String],
-  messageOfFalseRadioChoice: Option[String])
+case class RadioInvasiveStartSaAgentCode(hasSaAgentCode: Option[Boolean], saAgentCode: Option[String])
+case class RadioInvasiveTaxPayerOption(variant: Option[String], utr: Option[String], nino: Option[String])
 
 object RadioWithInput {
-  val confirmResponseForm: Form[RadioWithInput] = Form[RadioWithInput](
+  val validVariantsForTaxPayerOptionForm = Seq("utr", "nino", "cannotProvide")
```

Comment:
I wonder if an enum/sealed trait might be nice to use here for these different possible values. 
It should give the chance for some exhaustive pattern match checking (e.g. back in `CheckAgencyController.invasiveTaxPayerOption` on line 284).
It'd also keep a single source for all these different string values and stop the need to proliferate them around the codebase. That would help us maintain the code should they ever change in the future.